File: akka-java/src/main/java/org/hahn/spider/actor/FetchUrl.java
package org.hahn.spider.actor;

import akka.actor.UntypedActor;
import org.hahn.spider.message.Extract;
import org.hahn.spider.message.Fetch;

/**
 * Created by jianghan on 16/5/2.
 */
public class FetchUrl extends UntypedActor {

    @Override
    public void preStart() throws Exception {
    }

    @Override
    public void onReceive(Object msg) throws Exception {
        if (msg instanceof Fetch) {
            Fetch fetch = (Fetch) msg;
            System.out.println(fetch.toString());
            getContext().actorSelection("/user/extract").tell(new Extract(fetch.getUrl(), new byte[0]), getSelf());
            // getSelf().tell(new Extract(fetch.getUrl(), new byte[0]), getSelf());
            // getSender().tell(new Extract(fetch.getUrl(), new byte[0]), getSelf());
            // getSender().tell(new Save(fetch.getUrl()), getSelf());
        } else {
            unhandled(msg);
        }
    }

}

File: akka-scala/src/main/scala/hahn/learning/akka/spider/actor/FetchUrl.scala
package hahn.learning.akka.spider.actor

import akka.actor.{Actor, Props}
import hahn.learning.akka.spider.Message

/**
  * Created by jianghan on 16-5-26.
  */
class FetchUrl extends Actor {

  // val child = context.actorOf(Props[ExtractContent], name = "extract")

  def receive = {
    case Message.Fetch(url, tryNum) => {
      println(s"Fetch: ${url}\t${tryNum}")
      context.actorSelection("/user/extract") ! Message.Extract(url, Array[Byte]())
      // sender ! Message.Extract(url, Array[Byte]())
      // child ! Message.Extract(url, Array[Byte]())
      if (tryNum < 3) {
        self ! Message.Fetch(url, tryNum + 1)
      }
    }
    case _ => println("Error FetchUrl")
  }

}
